app/custom_auth.py

```diff
@@ -0,0 +1,57 @@
+from django.contrib import auth
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import PermissionDenied
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+class CustomSSOMiddleware:
+    """
+    A custom middleware that allows SSO via HTTP header passed in the request
+    The name of HTTP header is defined in the settings.py by the variable HEADER_CUSTOM_USER_NAME  
+
+    
+    (1) If HTTP header with the user name is not passed no further processing is done by the middleware.
+    (2) If HTTP header with the user name is passed then:
+        (2.1) Authenciate the user if already exist in the database. 
+        (2.2) Create user if does not exist in the database and then autenicate the user. 
+        (2.3) User is now signed in and session cookies are set.  
+    ...
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+     
+    def __call__(self, request):
+        
+        try:
+            username = request.META[settings.HEADER_CUSTOM_USER_NAME]
+           
+            if not User.objects.filter(username=username).exists() and settings.CREATE_USER:
+                create_user(username)
```

Comment:
[RemoteUserBackend](https://github.com/django/django/blob/3ee0834a4652fb9eb8b69e09c3541901739021de/django/contrib/auth/backends.py#L135) already has a flag to handle creating unknown users. We could create our own authentication backend which inherits from RemoteUserBackend and set the flag there.

Reply:
```auth.authenticate(remote_user=username)``` already support to creating unknow users. In fact that it is default behaviour when passing user name against ```remote_user``` parameter . Latest commit reflect this behaviour.